sql-plugin/src/main/scala/org/apache/spark/sql/rapids/datetimeExpressions.scala

```diff
@@ -177,6 +174,36 @@ case class GpuTimeSub(
       }
     }
   }
+
+  def intervalMath(us_s: Scalar, us: ColumnVector): ColumnVector
+}
+
+case class GpuTimeAdd(start: Expression,
+                      interval: Expression,
+                      timeZoneId: Option[String] = None)
+  extends GpuTimeMath(start, interval, timeZoneId) {
+
+  override def withTimeZone(timeZoneId: String): TimeZoneAwareExpression = {
+    copy(timeZoneId = Option(timeZoneId))
+  }
+
+  override def intervalMath(us_s: Scalar, us: ColumnVector): ColumnVector = {
+    us.add(us_s)
```

Comment:
in spark 3.1.0 TimeSub was removed and replaced with TimeAdd(l, -r) = TimeAdd(l, UnaryMinus(r)).  did you happen to run the 3.1.0 tests?  Its not a big deal here as we do have another issue for it but would be nice if this fixed both.

Reply:
Yes, I ran the tests for 3.1.0. The TimeSub test is xpassed for spark 3.1.0

I wanted to focus this PR on TimeAdd but I am happy to make the change.

Reply:
ok lets leave it separate